src/workerd/io/io-context.h

```diff
@@ -1275,18 +1276,30 @@ jsg::Promise<IoContext::MaybeIoOwn<addIoOwn, T>> IoContext::awaitIoImpl(
 
   typedef MaybeIoOwn<addIoOwn, T> Result;
 
-  v8::Isolate* isolate = getCurrentLock().getIsolate();
+  auto& lock = getCurrentLock();
+  v8::Isolate* isolate = lock.getIsolate();
+
   auto [ jsPromise, resolver ] = jsg::newPromiseAndResolver<Result>(isolate);
 
+  // In the then() continuations below we grab a reference to the jsg::AsyncContextFrame
+  // that is current when awaitIo is called and restore that frame when the promise is
+  // resolved. Why? The reasoning is straight forward: for promises, the context is captured
+  // when then is called, ensuring that when the continuation executes, it is running in that
+  // same context. When resolver.reject() or resolver.resolve() are called, v8 will pick up
```

Comment:
To be absolutely clear, this _only_ matters when the rejection is unhandled and an unhandled rejection event is fired, right? Because it affects the context seen in that handler? Can we be clearer about that in this comment?

Reply:
I tweaked it a bit but don't know if it's what you're looking for. Open to suggestions on rewording.